contentcuration/contentcuration/frontend/shared/utils/validation.js

```diff
@@ -28,6 +28,7 @@ import { ContentKindsNames } from 'shared/leUtils/ContentKinds';
  * Exercise with mastery model M of N must have valid N set
  * It must have at least one question
  * A question must have right answers
+ * No two questions should have the same order
```

Comment:
This function is not a validation function for developers. It validates rules that users should follow when creating content, and adding this rule might result in an error icon indicating that there is something that they need to fix, but users can't fix an issue when two items have the same order. They can just reorder items. Running into such a situation should not be possible. It might make sense as a console warning for developers though.

Reply:
That makes sense to me.  That said, I'd feel better if we had a way of enforcing on a _data_ level that no two questions in an exercise have the same order, since as of right now the swap functionality depends on it.  Typically this would happen on the backend (and maybe it should), but with the indexeddb layer, it seems that we should also be validating stuff like this on the front end too.

It strikes me that we ought to have a way of detecting if the data has somehow gotten corrupted in this way, then throwing an error and healing the corruption in the data.

For example, even if we did manage to fix all of the bugs causing this to happen for now, but then later on another bug crept in or we had a regression, we might end up with users creating lots of question sets where two items have `order: 0` and having no way of detecting this.

One thing that complicates this right now is that items are reordered one at a time, meaning that there are going to be some cases where two questions could have the same `order` temporarily, even if after an operation is finished, they'll all have unique orders.